Sources/FigmaExportCore/Processor/AssetsProcessor.swift

```diff
@@ -88,6 +88,197 @@ public struct ImagesProcessor: AssetsProcessable {
+                                      dark: [AssetType],
+                                      lightHighContrast: [AssetType],
+                                      darkHighContrast: [AssetType]) -> ProcessingPairResult {
+        // Error checks
+        var errors = ErrorGroup()
+        // CountMismatch
+        if light.count < dark.count {
+            errors.all.append(AssetsValidatorError.countMismatch(light: light.count, dark: dark.count))
+        }
+        // FoundDuplicate
+        let lightSet: Set<AssetType> = foundDuplicate(assets: light, errors: &errors)
+        let darkSet: Set<AssetType> = foundDuplicate(assets: dark, errors: &errors)
+        let lightHCSet: Set<AssetType> = foundDuplicate(assets: lightHighContrast, errors: &errors)
+        let darkHCSet: Set<AssetType> = foundDuplicate(assets: darkHighContrast, errors: &errors)
+        // AssetNotFoundInLightPalette
+        checkSubtracting(firstAsset: lightSet, secondAsset: darkSet, errors: &errors)
+        checkSubtracting(firstAsset: lightSet, secondAsset: lightHCSet, errors: &errors)
+        checkSubtracting(firstAsset: darkSet, secondAsset: darkHCSet, errors: &errors)
+
+        // DescriptionMismatch
+        lightSet.forEach { asset in
+            if let platform = asset.platform,
+               let dark = darkSet.first(where: { $0.name == asset.name }),
+               dark.platform != platform {
+                let error = AssetsValidatorError.descriptionMismatch(
+                    assetName: asset.name,
+                    light: platform.rawValue,
+                    dark: dark.platform?.rawValue ?? "")
+                errors.all.append(error)
+            }
+        }
+
+        // Return failure
+        guard !errors.all.isEmpty else { return .failure(errors) }
+        // Warning checks
+        var warning: AssetsValidatorWarning?
+        // LightAssetNotFoundInDarkPalette
+        let lightElements = lightSet.subtracting(darkSet)
+        if !lightElements.isEmpty { warning = .lightAssetsNotFoundInDarkPalette(assets: lightElements.map { $0.name }) }
+        // LightHCAssetNotFoundInDarkHCPalette
+        let lightHCElements = lightHCSet.subtracting(darkHCSet)
+        if !lightHCElements.isEmpty { warning = .lightHCAssetsNotFoundInDarkHCPalette(assets: lightHCElements.map { $0.name }) }
+
+        let pairs = makeSortedAssetPairs(lightSet: lightSet, darkSet: darkSet, lightHCSet: lightHCSet, darkHCSet: darkHCSet)
+        return .success(pairs, warning: warning)
+    }
+
+    private func checkSubtracting(firstAsset: Set<AssetType>, secondAsset: Set<AssetType>, errors: inout ErrorGroup) {
+        let elements = secondAsset.subtracting(firstAsset)
+        if !elements.isEmpty {
+            errors.all.append(AssetsValidatorError.darkAssetsNotFoundInLightPalette(assets: elements.map { $0.name }))
+        }
+    }
+
+    private func foundDuplicate(assets: [AssetType], errors: inout ErrorGroup) -> Set<AssetType> {
```

Comment:
Code in this method duplicated in the
`func validateAndMakePairs(light: [AssetType]) -> ProcessingPairResult {`
and
`func validateAndMakePairs(light: [AssetType], dark: [AssetType]) -> ProcessingPairResult {`
method. Could you please remove code duplication?